tests/e2e/test_e2e_ssp.py

```diff
@@ -111,33 +100,40 @@ def test_ssp_editing_e2e(
     tmp_repo_path = pathlib.Path(tmp_repo_str)
 
     args = setup_for_ssp(tmp_repo_path, test_prof, [test_comp_name], test_ssp_md)
+    remote_url = "http://localhost:8080/test.git"
+    repo.create_remote("origin", url=remote_url)
 
     # Create or generate the SSP
     if not skip_create:
-        index_path = os.path.join(tmp_repo_str, "ssp-index.json")
-        ssp_index = SSPIndex(index_path)
-        authored_ssp = AuthoredSSP(tmp_repo_str, ssp_index)
-        authored_ssp.create_new_default(
+        create_args: Dict[str, str] = {
+            "markdown-path": command_args["markdown-path"],
+            "branch": command_args["branch"],
+            "committer-name": command_args["committer-name"],
+            "committer-email": command_args["committer-email"],
+            "ssp-name": test_ssp_name,
+            "profile-name": test_prof,
+            "compdefs": test_comp_name,
+        }
+        command = build_test_command(
```

Comment:
The idiom of building and running a command might be further encapsulated ... some [inspiration](https://click.palletsprojects.com/en/8.1.x/testing/) eg. have a TestRunner abstraction with a build, invoke, etc methods might be handy testing command line invocation everywhere.